src/Akeneo/Bundle/StorageUtilsBundle/Elasticsearch/Cursor/Cursor.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+namespace Akeneo\Bundle\StorageUtilsBundle\Elasticsearch\Cursor;
+
+use Akeneo\Component\StorageUtils\Cursor\CursorInterface;
+use Akeneo\Component\StorageUtils\ElasticsearchClientInterface;
+use Akeneo\Component\StorageUtils\Repository\CursorableRepositoryInterface;
+
+/**
+ * Cursor to iterate on items
+ *
+ * @author    Marie Bochu <marie.bochu@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class Cursor implements CursorInterface
+{
+    /** @var ElasticsearchClientInterface */
+    protected $esClient;
+
+    /** @var CursorableRepositoryInterface */
+    protected $repository;
+
+    /** @var array */
+    protected $esQuery;
+
+    /** @var string */
+    protected $indexType;
+
+    /** @var array */
+    protected $items = [];
+
+    /** @var int */
+    protected $pageSize;
+
+    /**
+     * @param ElasticsearchClientInterface  $esClient
+     * @param CursorableRepositoryInterface $repository
+     * @param array                         $esQuery
+     * @param string                        $indexType
+     * @param int                           $pageSize
+     */
+    public function __construct(
+        ElasticsearchClientInterface $esClient,
+        CursorableRepositoryInterface $repository,
+        array $esQuery,
+        $indexType,
+        $pageSize
+    ) {
+        $this->repository = $repository;
+        $this->esClient = $esClient;
+        $this->esQuery = $esQuery;
+        $this->indexType = $indexType;
+        $this->pageSize = $pageSize;
+
+        $this->items = $this->getNextItems($esQuery);
```

Comment:
Yeah, it's weird why do we not to that on the first call of the current/or whatever ?